tdms/include/interpolate_Hfield.h

```diff
@@ -1,11 +1,49 @@
 /**
  * @file interpolate_Hfield.h
- * @brief Functions for interpolating the Electric field.
+ * @author William Graham (ccaegra@ucl.ac.uk)
+ * @brief Template functions for interpolating the magnetic field.
  */
 #pragma once
+#include "interpolation_methods.h"
+
+/* INTERPOLATION SCHEMES FOR THE MAGNETIC FIELD
+
+Unlike the E-field, the H-field components associated with Yee cell i,j,k are _not_ aligned with the centre of the Yee cells.
+Instead, the position of the field components (relative to the Yee cell centre is):
+
+Hx  | (0.0, 0.5, 0.5) .* (Dx, Dy, Dx)
+Hy  | (0.5, 0.0, 0.5) .* (Dx, Dy, Dz)
+Hz  | (0.5, 0.5, 0.0) .* (Dx, Dy, Dz)
+
+where Dx, Dy, Dz are the dimensions of the Yee cell.
+This requires us to interpolate twice to recover (any of the) field components at the centre of Yee cell i,j,k.
+
+Henceforth, let {a,b,c} = {x,y,z} be some 1-to-1 assignment of the co-ordinate axes to the labels a,b,c.
+The values Da, Db, Dc are the corresponding permutation of Dx, Dy, Dz.
+Suppose that we wish to interpolate the Ha field component to the centre a particular Yee cell.
+
+We will use the notation (a_i,b_j,c_k) for the Yee cell indices, although bear in mind that this does not reflect the order the indices appear in the code.
```

Comment:
@willGraham01 , could you add just a little more explanation here? I don't quite follow the notation. Is the logic below that you will always access Ha, and therefore, the field component of interest will be chosen by setting a (i.e., if interested in Hz, we assign a=z)? In this case it makes sense.

Reply:
Yep this was the intention - we always try and extract `Ha`. That's also why there are [comments like this](https://github.com/UCL/TDMS/blob/fbc348174285a3874dfdc5f9d5ee356c8a5511a2/tdms/include/interpolate_Hfield.h#L59) after each function is defined. If we are interested in `Hz`, as you say we use `a=z`. Then it doesn't really whether we choose `b=x, c=y` or `b=y, c=x` since we interpolate in each of these directions anyway. The reason for the abstract notation is just to highlight that whichever direction we choose for `a`, the two dimensions we interpolate in are the _other_ dimensions `b` and `c`.

Originally I had this comment at the start of each function, explicitly using Hx, Hy, and Hz and moving the roles around as necessary. That resulted in a huge file-bloat and a lot of repeated information, so we abstracted it and moved it outside the functions into one comment at the beginning of the file.